offchain-modules/lib/src/dapp/indexer/eth_indexer.rs

```diff
@@ -0,0 +1,260 @@
+use crate::dapp::indexer::db::{
+    create_eth_to_ckb_record, get_ckb_to_eth_record_by_eth_hash, get_latest_eth_to_ckb_record,
+    is_eth_to_ckb_record_exist, update_ckb_to_eth_record_status, EthToCkbRecord,
+};
+use crate::transfer::to_ckb::to_eth_spv_proof_json;
+use crate::util::config::ForceConfig;
+use crate::util::eth_util::{convert_hex_to_h256, Web3Client};
+use anyhow::{anyhow, Result};
+use ckb_hash::blake2b_256;
+use ckb_jsonrpc_types::Uint128;
+use ethabi::{Function, Param, ParamType};
+use log::info;
+use rusty_receipt_proof_maker::types::UnlockEvent;
+use rusty_receipt_proof_maker::{generate_eth_proof, parse_unlock_event, types::EthSpvProof};
+use shellexpand::tilde;
+use sqlx::MySqlPool;
+use std::ops::Add;
+use web3::types::U64;
+
+pub struct EthIndexer {
+    pub force_config: ForceConfig,
+    pub eth_client: Web3Client,
+    pub db: MySqlPool,
+}
+
+impl EthIndexer {
+    pub async fn new(
+        config_path: String,
+        network: Option<String>,
+        db_path: String,
+    ) -> Result<Self> {
+        let config_path = tilde(config_path.as_str()).into_owned();
+        let force_config = ForceConfig::new(config_path.as_str())?;
+        let eth_rpc_url = force_config.get_ethereum_rpc_url(&network)?;
+        let eth_client = Web3Client::new(eth_rpc_url);
+        let db = MySqlPool::connect(&db_path).await?;
+        Ok(EthIndexer {
+            force_config,
+            eth_client,
+            db,
+        })
+    }
+
+    pub async fn start(&mut self) -> Result<()> {
+        let record_option = get_latest_eth_to_ckb_record(&self.db).await?;
+        let mut start_block_number;
+        if record_option.is_some() {
+            start_block_number = U64::from(record_option.unwrap().block_number);
+        } else {
+            start_block_number = self.eth_client.client().eth().block_number().await?;
+        }
+
+        loop {
+            let block = self.eth_client.get_block(start_block_number.into()).await;
+            if block.is_err() {
+                info!("waiting for new block.");
+                tokio::time::delay_for(std::time::Duration::from_secs(3)).await;
+                continue;
+            }
+            let txs = block.unwrap().transactions;
+            for tx_hash in txs {
+                let hash = hex::encode(tx_hash);
+                if !is_eth_to_ckb_record_exist(&self.db, &hash).await? {
+                    self.handle_lock_event(hash.clone(), start_block_number.as_u64())
+                        .await?;
+                }
+            }
```

Comment:
How about filter all the lock+unlock txs, and insert them to db once (atomic) ? Because eth tx relayer require this property: when eth tx relayer filter txs from indexer-table by block number, it should be integrated

Reply:
https://github.com/LeonLi000/force-bridge-eth/commit/8083cde008c76f89158844901a52592c3ede5239#diff-d1833c71df3dc5e85e67ceefdd801c93bb23e12406de2c57a6ca02ca41edd772R88